xds/internal/balancer/clusterimpl/picker.go

```diff
@@ -156,6 +156,12 @@ func (d *picker) Pick(info balancer.PickInfo) (balancer.PickResult, error) {
 		return pr, err
 	}
 
+	if info.Ctx != nil {
+		if labels := stats.GetLabels(info.Ctx); labels != nil && labels.TelemetryLabels != nil {
```

Comment:
This is the second call to `if info.Ctx != nil { ... GetLabels() ...` in this function..can we try to combine so we aren't duplicating this logic?

maybe

```go
func telLabels(ctx context.Context) map[string]string {
	if ctx == nil {
		return nil
	}
	labels := stats.GetLabels(info.Ctx)
	if labels == nil {
		return nil
	}
	return labels.TelemetryLabels
}

// ...

if labels := telLabels(info.Ctx); labels != nil {
	labels["grpc.lb.locality"] = lIDStr
}
```

Reply:
Ah yeah I like refactoring this into a helper. I wanted to combine the writes, but I think where the two writes (unconditional for xDS Labels and here for locality) are the right place to put them.